code/player/Player.cs

```diff
@@ -2,16 +2,30 @@
 
 using TTTReborn.Gamemode;
 using TTTReborn.Player.Camera;
+using TTTReborn.Roles;
 
 namespace TTTReborn.Player
 {
     public partial class TTTPlayer : Sandbox.Player
     {
-        public enum RoleType { None, Innocent, Detective, Traitor }
         public PlayerCorpse PlayerCorpse { get; set; }
 
-        [Net, Local]
-        public RoleType Role { get; set; }
+        public BaseRole Role {
+            set
+            {
+                role = value;
+
+                if (IsServer)
+                {
+                    SetRole(To.Single(this), role.Name);
+                }
+            }
+            get {
+                return role;
+            }
+        }
+
+        private BaseRole role = new NoneRole();
```

Comment:
Unless I'm misunderstanding, where do we keep track of the players role on the server?

Reply:
Currently, the role on the server is the role the client has. No client is aware of the role of other players, but I thought that's something for another PR as it gonna be complex (storing information the client is aware of, displaying other roles to the clients than they actually have, etc.).